src/main/java/com/GeorgesServer/app/SocketFactory.java

```diff
@@ -0,0 +1,58 @@
+package com.GeorgesServer.app;
+
+import java.io.*;
+import java.net.ServerSocket;
+import java.net.Socket;
+
+public class SocketFactory implements ISocketFactory {
```

Comment:
This class is doing too much and it's not implementing the factory pattern correctly.